src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1461,8 +1470,14 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
                 // User does not want to launch activity
                 Logger.e(TAG, "Prompt is not allowed and failed to get token:", request.getLogInfo() + " " + errorInfo,
                         ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED);
-                callbackHandle.onError(new AuthenticationException(
-                        ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, request.getLogInfo() + " " + errorInfo));
+                if (authResult != null && authResult.getErrorCode() != null) {
+                    callbackHandle.onError(new AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, 
```

Comment:
So as a result we actually should sign out users only if we've got internal cause and message of this internal exception contains invalid_grant...
ok, I'm glad that we know it, but I'm sorry for other users of this lib...how the hell they should know such thing!